lib/index_documents.rb

```diff
@@ -0,0 +1,54 @@
+require_relative "../app"
+require 'gds_api/publishing_api_v2'
+
+class IndexDocuments
+  def process(message)
+    base_path = document_base_path(message)
+    topic_content_ids = topic_content_ids(message)
+
+    unless topic_content_ids.nil? # no topics field in links, don't update
+      topic_base_paths = topic_content_ids.map { |content_id| base_path(content_id) }
+      update_index(base_path, topic_base_paths)
+    end
+
+    message.ack
+  rescue StandardError => e
+    Airbrake.notify_or_ignore(
+      error: e,
+      payload: message.payload,
+      headers: message.headers,
+      delivery_info: message.delivery_info
+    )
+    message.retry
+  end
+
+  private
+
+  def document_base_path(message)
+    message.payload["base_path"]
+  end
+
+  def topic_content_ids(message)
+    message.payload["links"].try(:[], "topics")
```

Comment:
Ah.  From the documentation, it looks like that will then return `nil` instead of `[]`, which will then fail when the return value is passed to `map`.

Would `message.payload.fetch("links", {}).fetch("topics", [])` be better?